### PR TITLE
Replace Travis with GHA

### PR DESCRIPTION
**Pull request by PenghaiZhang**
Initial work for replacing Travis with GHA.

### CODE REVIEW

**Comment by edalex-ian** [comment]
I note there's no release element here yet - whereas the Travis build had that feature.

The original official action seemed to be https://github.com/actions/create-release . You'll note it is now archived but links to four others.

This one looks good: https://github.com/softprops/action-gh-release

**Comment by PenghaiZhang** [comment]
image

Publish releases by GHA. Tested on my folk repo.

**Comment by edalex-ian** [comment]
Looks very promising. Let me know when you want it reviewed. :+1:

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            ${{ runner.os }}-gradle-
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 8
+
+      - name: Build with Gradle
+        run: ./gradlew build
```
Comment (ChristianMurphy) [inline .github/workflows/ci.yaml:40]: Does the `build` task also run the full test suite? (I think it does, just want to confirm)

**Review by ChristianMurphy — APPROVED** [review]

**Comment by PenghaiZhang** [comment]
> Looks very promising. Let me know when you want it reviewed.

It's ready for review now :smile:

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (PenghaiZhang) [reply .github/workflows/ci.yaml:40]: Yes I think so.

> By default, Gradle will run all tests that it detects, which it does by inspecting the compiled test classes. This detection uses different criteria depending on the test framework used.

Comment (edalex-ian) [reply .github/workflows/ci.yaml:40]: I confirmed by looking at the build output and it did report compile test classes and running tests.

That said, unfortunately there is only one test. :'(

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Save artefacts
+        uses: actions/upload-artifact@v2.2.3
+        with:
+          name: Artefacts
+          path: build/distributions
```
Comment (edalex-ian) [inline .github/workflows/ci.yaml:46]: After the talk above about the tests, I wonder if perhaps we should save also the test report.

**Review by edalex-ian — APPROVED** [review]
LGTM @PenghaiZhang . And I love the nice simple method of creating releases, that's and improvement and if it works out well we should look in the future to add it to the oEQ build.

I did question whether we should also save the test reports, so if you're willing to add that it could be good. But overall, feel free to merge when you're ready. ;)

**Comment by PenghaiZhang** [comment]
Coverage report is now also saved.

image

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Save Coverage Report
```
Comment (edalex-ian) [inline .github/workflows/ci.yaml:48]: Not actually a Coverage report, just a Test report. But ah well, not to worry. :)

**Pull request merged**